ext/OceananigansReactantExt/Architectures.jl

```diff
@@ -0,0 +1,25 @@
+module Architectures
+
+using Reactant
+using Oceananigans
+
+import Oceananigans.Architectures: device, architecture, array_type, on_architecture, unified_array, device_copy_to!
+
+architecture(::Reactant.AnyConcreteRArray) = ReactantState
+architecture(::Reactant.AnyTracedRArray) = ReactantState
+
+array_type(::ReactantState) = ConcreteRArray
+
+arch_array(::CPU, a::ReactantState) = Array(a)
```

Comment:
```suggestion
arch_array(::CPU, a::ReactantState) = Array(a)
```
this is deprecated, I think we can just remove it completely

Reply:
@simone-silvestri might be best if you change this yourself